### PR TITLE
Transfer all bytes with Guava instead of parts with IOUtils.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
A user report that the event stream would stutter. He found that the
change to Apache IOUtils is not reading the complete input stream and
also buffers the input before it's copied. We switch to Guava
ByteStreams.copy which copies all bytes.

JIRA issues: MARATHON-7919

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala [hunk 1]
```diff
@@ -294,13 +294,13 @@ class JavaUrlConnectionRequestForwarder @Inject() (
...
-          Try(leaderConnection.getErrorStream.close())
+          Closeables.closeQuietly(leaderConnection.getInputStream())
+          Closeables.closeQuietly(leaderConnection.getErrorStream())
         }
       }
     } finally {
-      Try(request.getInputStream.close())
-      Try(response.getOutputStream.close())
+      Closeables.closeQuietly(request.getInputStream())
+      Closeables.close(response.getOutputStream(), true)
```
Comment (zen-dog) [inline src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala]: Why is the output stream not closed quietly?

Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/LeaderProxyFilter.scala]: It is. The parameter is called `swallowIOException`. There is not `closeQuietly` for output streams.

**Comment by alenkacz** [comment]
I reviewed but since there is still ongoing discussion about whether it helps or not I will wait with my 👍 :) https://jira.mesosphere.com/browse/MARATHON-7919

**Comment by jeschkies** [comment]
> I reviewed but since there is still ongoing discussion about whether it helps or not I will wait with my 👍 

Yep, could you ask for changes though? I'm kind of surprised that this patch would not work.

**Comment by jeschkies** [comment]
> Based on https://jira.mesosphere.com/browse/MARATHON-7919 I am not sure if we should merge this. Maybe we need to test this or verify in a way that it helped... ?

I agree. Let's drop it for now.

**Pull request closed without merging**